src/webviews/settings/index.ts

```diff
@@ -303,6 +303,7 @@ export class SettingsUI {
               if (!data.privateKeyPath?.trim()) {
                 if (connection.privateKeyPath?.trim()) {
                   data.privateKeyPath = connection.privateKeyPath;
+                  context.secrets.delete(`${name}_password`);
```

Comment:
It's an async call, we should await.
```suggestion
                  await context.secrets.delete(`${name}_password`);
```